src/Chess/Game/ChessGame.php

```diff
@@ -1367,12 +1365,58 @@ public function inClaimableDraw()
      */
     public function inRepetitionDraw()
     {
-        $fen = $this->renderFen(false);
-        if (isset($this->_allFENs[$fen]) && $this->_allFENs[$fen] >= 3) {
-            return true;
+        if (3 > $this->_halfMoves) {
+            return false;
         }
 
-        return false;
+        $samePositions = array_filter(
+            array_values(array_count_values($this->fens)),
+            function ($value) {
+                return 3 <= $value;
+            }
+        );
+
+        return (bool) $samePositions;
+    }
+
+
+    /**
+     * The game is drawn without either player making a claim if the same
+     * position occurs on the board after five consecutive pairs of moves
+     * for each player ("five-fold repetition")
+     *
+     * @return bool
+     */
+    public function inFiveFoldRepetitionDraw()
+    {
+        if (20 > $this->_halfMoves) {
+            return false;
+        }
+
+        $last5Moves = array_slice($this->fens, -20, 20);
+
+        $samePositions = array_filter(
+            array_values(array_count_values($last5Moves)),
+            function ($value) {
+                return 5 <= $value;
+            }
+        );
+
+        if (!$samePositions) {
+            return false;
+        }
+
+        $odd = array();
+        $even = array();
+        foreach ($last5Moves as $key => $fen) {
+            if (0 === $key % 2) {
+                $even[] = $fen;
+            }
+            else {
+                $odd[] = $fen;
+            }
+        }
+        return 2 === count(array_unique($even)) && 2 === count(array_unique($odd));
```

Comment:
not really sure i understand this. can you put a comment here in the code that explains this logic?